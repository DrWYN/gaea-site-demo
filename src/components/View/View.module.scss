.view {
  -webkit-transform: translateZ(0px);
  -moz-transform: translateZ(0px);
  -ms-transform: translateZ(0px);
  -o-transform: translateZ(0px);
  transform: translateZ(0px);
}