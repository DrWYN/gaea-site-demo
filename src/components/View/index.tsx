import React from 'react';

type IProps = React.HTMLAttributes<any> & { 
  testID?: string;
  accessible?: boolean;
 }

class View extends React.Component<IProps, any> {
  constructor(props: IProps) {
    super(props);
  }

  render() {
    const {children, testID = '', ...other} = this.props;
    return (
      <div
        {...other} 
        data-testid={testID}>
      { children }
      </div>
    );
  }
}

export default View;
