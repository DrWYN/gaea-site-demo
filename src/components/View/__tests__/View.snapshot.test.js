import React from 'react';
import renderer from 'react-test-renderer';
import View from '../index';

describe('<View />', () => {
  it('no props', () => {
    const component = renderer.create(<View />);
    
    let snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('props testID', () => {
    const component = renderer.create(<View testID="test"/>);

    let snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
})
