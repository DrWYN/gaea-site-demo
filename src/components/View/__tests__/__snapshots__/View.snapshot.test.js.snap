// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<View /> no props 1`] = `
<div
  data-testid=""
/>
`;

exports[`<View /> props testID 1`] = `
<div
  data-testid="test"
/>
`;
